pkg/nsx/services/vpc/store.go

```diff
@@ -74,7 +75,10 @@ func (vs *VPCStore) Apply(i interface{}) error {
 	return nil
 }
 
-func (vs *VPCStore) GetVPCsByNamespace(ns string) []*model.Vpc {
+func (vs *VPCStore) GetVPCsByNamespace(ns *v1.Namespace) []*model.Vpc {
+	if ns == nil {
```

Comment:
when you use the GetVPCsByNamespace, it seems you didn't check if the result is nil, just iterate. How about also return empty []*model.Vpc if ns== nil?

Reply:
The ret is nil in `var ret []*model.Vpc`, if no value is assigned to this value. It’s equal to nil.